quic/s2n-quic-core/src/recovery/congestion_controller/fuzz_target.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    packet::number::PacketNumberSpace,
+    path::MINIMUM_MTU,
+    random,
+    recovery::{
+        bbr::BbrCongestionController, CongestionController, CubicCongestionController, RttEstimator,
+    },
+    time::{testing::Clock, Clock as _, Timestamp},
+};
+use bolero::{check, generator::*};
+use core::time::Duration;
+use std::collections::VecDeque;
+
+struct SentPacketInfo<PacketInfo> {
+    sent_bytes: u16,
+    time_sent: Timestamp,
+    cc_packet_info: PacketInfo,
+}
+
+#[derive(Debug, TypeGenerator)]
+enum Operation {
+    IncrementTime {
+        /// The milli-second value by which to increase the timestamp
+        millis: u16,
+    },
+    PacketSent {
+        #[generator(1..=255)]
+        count: u8,
+        #[generator(1200..=9000)]
+        bytes_sent: u16,
+        app_limited: Option<bool>,
+    },
+    RttUpdated {
+        #[generator(1..=2000)]
+        millis: u64,
```

Comment:
It's usually better for the fuzzer to use the minimal type for your range, otherwise you're consuming a larger portion of the buffer but not doing anything with it. It would be nice to improve this in bolero though.

```suggestion
        millis: u16,
```

Reply:
Oh ignore me: I've already implemented it 😄 

https://github.com/camshaft/bolero/blob/7cd069b181da07c85a133d0ed6c85b8f05221949/bolero-generator/src/uniform.rs#L71